mods/tools/replacer.lua

```diff
@@ -0,0 +1,237 @@
+            })
+        end
+        return nil
+    end
+    -- in survival mode, the player has to provide the node he wants to place
+    if (not (minetest.settings:get_bool("creative_mode")) and not (minetest.check_player_privs(name, {
+        creative = true
+    }))) then
+        -- players usually don't carry dirt_with_grass around it's safe to assume normal dirt here
+        -- fortunately, dirt and dirt_with_grass does not make use of rotation
+        if (daten[1] == "default:dirt_with_grass") then
+            daten[1] = "default:dirt"
+            item["metadata"] = "default:dirt 0 0"
+        end
+
+        -- does the player carry at least one of the desired nodes with him?
+        if (not (user:get_inventory():contains_item("main", daten[1]))) then
+            ccore.notify(name, "You have no further '" .. (daten[1] or "?") .. "'. Replacement failed.")
+            return nil
+        end
+        -- give the player the item by simulating digging if possible
+        if (node.name ~= "air" and node.name ~= "ignore" and node.name ~= "default:lava_source" and node.name ~=
+            "default:lava_flowing" and node.name ~= "default:river_water_source" and node.name ~=
+            "default:river_water_flowing" and node.name ~= "default:water_source" and node.name ~=
+            "default:water_flowing") then
+            minetest.node_dig(pos, node, user)
+            local digged_node = minetest.get_node_or_nil(pos)
+            if (not (digged_node) or digged_node.name == node.name) then
+                ccore.notify(name, "Replacing '" .. (node.name or "air") .. "' with '" .. (item["metadata"] or "?") ..
+                    "' failed.\nUnable to remove old node.")
+                return nil
+            end
+        end
+        -- consume the item
+        user:get_inventory():remove_item("main", daten[1] .. " 1")
+    end
+
+    minetest.add_node(pos, {
+        name = daten[1],
+        param1 = daten[2],
+        param2 = daten[3]
+    })
+    return nil -- no item shall be removed from inventory
+end
+-- protection checking from Vanessa Ezekowitz' homedecor mod
+-- see http://forum.minetest.net/viewtopic.php?pid=26061 or https://github.com/VanessaE/homedecor for details!
+replacer.node_is_owned = function(pos, placer)
+    if (not (placer) or not (pos)) then
+        return true
+    end
+    local pname = placer:get_player_name()
+    if (type(minetest.is_protected) == "function") then
+        local res = minetest.is_protected(pos, pname)
+        if (res) then
+            ccore.notify(pname, "Cannot replace node. It is protected.")
+        end
+        return res
+    end
+    return false
+end
+
+-- Handle mode setting/getting
+replacer.set_mode = function(player, mode_name)
+    if mode_name ~= "legacy" and mode_name ~= "paint" then
+        return ccore.notify(player:get_player_name(), "Invalid replacer mode!")
+    end
+    local meta = player:get_meta()
+    meta:set_string('replacer_mode', mode_name)
```

Comment:
same comment as hud above.
To prevent accidental overrides, `<modname>:<name>` is preferred
Especially when dealing with player meta, something easily accessible